ecs-cluster/keycloak.tf

```
@@ -5,8 +5,8 @@ locals {
   keycloak-hostname = var.keycloak-hostname == "" ? aws_lb.keycloak.dns_name : var.keycloak-hostname
 
   vpc_id          = var.vpc-id == "" ? module.vpc[0].vpc_id : var.vpc-id
-  public_subnets  = var.public-subnets == [] ? module.vpc[0].public_subnets : var.public-subnets
-  private_subnets = var.private-subnets == [] ? module.vpc[0].private_subnets : var.private-subnets
+  public_subnets  = length(var.public-subnets) == 0 ? module.vpc[0].public_subnets : var.public-subnets
+  private_subnets = length(var.private-subnets) == 0 ? module.vpc[0].private_subnets : var.private-subnets
```

Feel free to decline but you can use coalesce

```suggestion
  public_subnets  = coalesce(module.vpc[0].public_subnets, var.public-subnets)
  private_subnets = coalesce(module.vpc[0].private_subnets, var.private-subnets)
```

---

`coalesce` is for strings, I tried [`coalescelists`](https://developer.hashicorp.com/terraform/language/functions/coalescelist) in [4c92a0a](https://github.com/hic-infra/ecs-keycloak/pull/16/commits/4c92a0aa7d647bcd80060e8e4578561fd5c64b18) but it fails if `module.vpc` doesn't exist because it doesn't use short-circuiting so I reverted it 😞.

---

coalesce would work if the defaults for those variables were null

```
> coalesce(null, ["b", "c", "d"])
[
  "b",
  "c",
  "d",
]
```

but I guess you might have the same issue :D oh well